packages/syft/src/syft/service/code/user_code.py

```diff
@@ -744,31 +744,49 @@ def byte_code(self) -> PyCodeObject | None:
         return compile_byte_code(self.parsed_code)
 
     @property
-    def assets(self) -> list[Asset]:
-        # relative
-        from ...client.api import APIRegistry
-
-        api = APIRegistry.api_for(self.node_uid, self.syft_client_verify_key)
-        if api is None:
-            return SyftError(message=f"You must login to {self.node_uid}")
+    def assets(self) -> list[Asset] | SyftError:
+        if not self.input_policy:
```

Comment:
This will be an issue with custom input policies, they are None until the code is approved.

Alternative is to look at input_policy_input_kwargs, but I don't know if we enforce some structure on those when using custom input policies